subsys/spm/spm.c

```diff
@@ -29,6 +29,13 @@
 #define NON_SECURE_APP_ADDRESS DT_FLASH_AREA_IMAGE_0_NONSECURE_OFFSET_0
 #endif /* USE_PARTITION_MANAGER */
 
+#if defined(CONFIG_SPM_DEBUG)
+#define SPM_DEBUG true
+#warning "SPM debug is enabled, so some security is disabled"
```

Comment:
This should probably be rephrased. 
Maybe:

SPM debug is enabled so this should not be used in production since parts of the security features of the SPM is disabled.

Reply:
Proposal: "!!! CONFIG_SPM_DEBUG is enabled, not intended for end-product !!!"

Reply:
I think 'warning' is strong enough so we can do without the exclamations :)

```suggestion
#warning "SPM debug is enabled, so some security features are disabled. Do not use this configuration in the final product."
```

Reply:
Changed